src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/LuceneDocumentConverterBase.cs

```diff
@@ -191,7 +193,21 @@ protected IEnumerable<AbstractField> GetRegularFields(IndexField field, object v
         private IEnumerable<AbstractField> GetComplexObjectFields(string path, Field.Store storage, BlittableJsonReaderObject val, Field.Index indexing, Field.TermVector termVector)
         {
             yield return GetOrCreateField(path + ConvertToJsonSuffix, TrueString, null, null, storage, Field.Index.NOT_ANALYZED_NO_NORMS, Field.TermVector.NO);
-            yield return GetOrCreateField(path, null, null, val.GetJsonReader(), storage, indexing, termVector);
+
+            StreamReader inMemoryComplexObject;
+            if (_complexInMemoryObjects.TryGetValue(path, out inMemoryComplexObject) == false)
```

Comment:
Why are we doing it per path? What happen if we have an array, with multiple values in the same path?